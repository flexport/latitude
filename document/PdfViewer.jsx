/**
 * TEAM: frontend_infra
 *
 * @flow
 */

import React, {useEffect, useMemo} from "react";

type Props = {|+pdf: File | string|};

/**
 * @short Previews PDF documents from `File` objects or URLs
 * @brandStatus V2
 * @status Beta
 * @category Documents
 *
 * This component uses the browser's native PDF renderer by opening the PDF in an `iframe`.
 */
export default function PdfViewer({pdf}: Props) {
  const url = useMemo(
    () => (pdf instanceof File ? URL.createObjectURL(pdf) : pdf),
    [pdf]
  );
  useEffect(() => {
    if (pdf instanceof File) URL.revokeObjectURL(url);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [pdf]);
  return (
    <iframe
      title={pdf instanceof File ? pdf.name : pdf}
      src={url}
      style={{width: "100%", height: "100%"}}
    />
  );
}
