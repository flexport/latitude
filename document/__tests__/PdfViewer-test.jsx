/**
 * TEAM: frontend_infra
 *
 * @flow
 */
import React from "react";
import {shallow} from "enzyme";
import PdfViewer from "../PdfViewer";

describe("DocumentUploader", () => {
  it("renders", () => {
    expect(shallow(<PdfViewer pdf="" />)).toMatchSnapshot();
  });
});
