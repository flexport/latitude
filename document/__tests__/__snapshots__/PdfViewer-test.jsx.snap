// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentUploader renders 1`] = `
<iframe
  src=""
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  title=""
/>
`;
